kafka-connect-http/src/main/java/com/github/castorm/kafka/connect/http/response/timestamp/RegexTimestampParser.java

```diff
@@ -0,0 +1,70 @@
+package com.github.castorm.kafka.connect.http.response.timestamp;
+
+/*-
+ * #%L
+ * Kafka Connect HTTP Plugin
+ * %%
+ * Copyright (C) 2020 CastorM
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.github.castorm.kafka.connect.http.response.timestamp.spi.TimestampParser;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@RequiredArgsConstructor
+@Slf4j
+public class RegexTimestampParser implements TimestampParser {
+
+  private final Function<Map<String, ?>, RegexTimestampParserConfig> configFactory;
+  private DateTimeFormatter timestampFormatter;
+  private Pattern pattern;
+  private String timestampPattern;
+  private String zone;
+
+  @Override
+  public Instant parse(String timestamp) {
+    Matcher matcher = pattern.matcher(timestamp);
+    String extractedTimestamp;
+    matcher.find();
+    extractedTimestamp = matcher.group(1);
+    if(!timestampPattern.equals("L")) {
+      return OffsetDateTime.parse(extractedTimestamp, timestampFormatter).toInstant();
+    }
+    return Instant.ofEpochMilli(Long.valueOf(extractedTimestamp));
```

Comment:
What would you think of delegating the timestamp extracted with the regex to another `TimestampParser`. You would then just chain this `RegexTimestampParser` with any other parser of your choice.

I believe that would simplify this parser, reducing its responsibilities and it would extend its possibilities by allowing its composition with existing and future parsers.

Any thoughts?

Reply:
Yes, that would be a better approach. I have made the changes to delegate it to another parser after extracting it from the regex pattern.